src/server/string_family.cc

```diff
@@ -658,19 +648,11 @@ OpStatus SetCmd::SetExisting(const SetParams& params, PrimeIterator it, ExpireIt
     db_slice.SetMCFlag(op_args_.db_cntx.db_index, it->first.AsRef(), params.memcache_flags);
   }
 
+  db_slice.RemoveFromTiered(it, op_args_.db_cntx.db_index);
   // overwrite existing entry.
   prime_value.SetString(value);
   DCHECK(!prime_value.HasIoPending());
 
-  if (value.size() >= kMinTieredLen) {  // external storage enabled.
-    // TODO: if UnloadItem can block the calling fiber, then we have the bug because then "it"
-    // can be invalid after the function returns and the functions that follow may access invalid
-    // entry.
-    if (shard->tiered_storage()) {
-      shard->tiered_storage()->ScheduleOffload(op_args_.db_cntx.db_index, it);
```

Comment:
we should not ScheduleOffload for existing entries as they are hot and should remain in memory